dispatcher.go

```diff
@@ -224,3 +256,27 @@ func (d dispatcher) Stop() error {
 
 	return nil
 }
+
+// getUniqueOutbounds helps ensure we only start/stop the same outbound once
+func (d dispatcher) getUniqueOutbounds() []transport.Outbound {
+	seen := make(map[transport.Outbound]struct{})
+	unique := []transport.Outbound{}
```

Comment:
Minor: You know the approximate size for both things: `len(d.outbounds)*2`. Use that in the `make` call.

```
seen := make(..., len(d.outbounds)*2)
unique := make([]transport.Outbound, 0, len(d.outbounds)*2)
```